### PR TITLE
Add module-info.java files to gRPC modules (issue #759)

### PR DESCRIPTION
**Pull request by thegridman**

### CODE REVIEW

File: grpc/io.grpc/pom.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    limitations under the License.
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
```
Comment (romain-grecourt) [inline grpc/io.grpc/pom.xml]: The indentation seems to be 2 instead of 4 here.

**Comment by thegridman** [comment]
The reason for repackaging is because the grpc-java project is not compatible with Java 9 modules due to having split package issues. The grpc-java project are not fixing this and the suggested workaround is to do what we have done here and repackage some of the grpc-java jars into a single jar and add a module-info.

File: grpc/pom.xml [hunk 2]
```diff
@@ -34,10 +34,11 @@
...
-    <modules>
-        <module>core</module>
-        <module>server</module>
-        <module>client</module>
-        <module>metrics</module>
-    </modules>
+  <modules>
+    <module>io.grpc</module>
+    <module>core</module>
+    <module>server</module>
```
Comment (romain-grecourt) [inline grpc/pom.xml]: This looks like 2 spaces indent, it should be 4.

**Comment by thegridman** [comment]
Can someone look at whether this can be merged please.

**Comment by romain-grecourt** [comment]
@barchetta ping.

**Pull request merged**